epochStart/bootstrap/syncValidatorStatus_test.go

```diff
@@ -95,6 +99,148 @@ func TestSyncValidatorStatus_NodesConfigFromMetaBlock(t *testing.T) {
+
+	metaBlock := &block.MetaBlock{
+		Nonce: 10,
+		Epoch: 1,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: mbHeaderHash1,
+				Type: block.TxBlock,
+			},
+			{
+				Hash: mbHeaderHash2,
+				Type: block.PeerBlock,
+			},
+		},
+	}
+
+	expectedBody := &block.Body{
+		MiniBlocks: []*block.MiniBlock{
+			{
+				ReceiverShardID: 1,
+				SenderShardID:   0,
+			},
+		},
+	}
+
+	svs, _ := NewSyncValidatorStatus(args)
+
+	wasCalled := false
+	svs.nodeCoordinator = &mock.NodesCoordinatorStub{
+		EpochStartPrepareCalled: func(metaHdr data.HeaderHandler, body data.BodyHandler) {
+			wasCalled = true
+			assert.Equal(t, metaBlock, metaHdr)
+			assert.Equal(t, expectedBody, body)
+		},
+	}
+	svs.miniBlocksSyncer = &epochStartMocks.PendingMiniBlockSyncHandlerStub{
+		SyncPendingMiniBlocksCalled: func(miniBlockHeaders []data.MiniBlockHeaderHandler, ctx context.Context) error {
+			return nil
+		},
+		GetMiniBlocksCalled: func() (map[string]*block.MiniBlock, error) {
+			return map[string]*block.MiniBlock{
+				string(mbHeaderHash2): {
+					ReceiverShardID: 1,
+					SenderShardID:   0,
+				},
+			}, nil
+		},
+	}
+
+	err := svs.processValidatorChangesFor(metaBlock)
+	require.NoError(t, err)
+	assert.True(t, wasCalled)
+}
+
+func TestSyncValidatorStatus_findPeerMiniBlockHeaders(t *testing.T) {
+	t.Parallel()
+
+	metaBlock := &block.MetaBlock{
+		Nonce: 37,
+		Epoch: 0,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: []byte("mb-hash1"),
+				Type: block.TxBlock,
+			},
+			{
+				Hash: []byte("mb-hash2"),
+				Type: block.PeerBlock,
+			},
+		},
+	}
+
+	expectedMbHeaders := []data.MiniBlockHeaderHandler{
+		&block.MiniBlockHeader{
+			Hash: []byte("mb-hash2"),
+			Type: block.PeerBlock,
+		},
+	}
+
+	mbHeaderHandlers := findPeerMiniBlockHeaders(metaBlock)
+	require.Equal(t, expectedMbHeaders, mbHeaderHandlers)
+}
+
+func TestSyncValidatorStatus_getPeerBlockBodyForMeta(t *testing.T) {
+	t.Parallel()
+
+	args := getSyncValidatorStatusArgs()
+
+	mbHeaderHash1 := []byte("mb-hash1")
+	mbHeaderHash2 := []byte("mb-hash2")
+
+	metaBlock := &block.MetaBlock{
+		Nonce: 37,
+		Epoch: 0,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{
+				Hash: mbHeaderHash1,
+				Type: block.TxBlock,
+			},
+			{
+				Hash: mbHeaderHash2,
+				Type: block.PeerBlock,
+			},
+		},
+	}
+
+	svs, _ := NewSyncValidatorStatus(args)
+	svs.miniBlocksSyncer = &epochStartMocks.PendingMiniBlockSyncHandlerStub{
+		SyncPendingMiniBlocksCalled: func(miniBlockHeaders []data.MiniBlockHeaderHandler, ctx context.Context) error {
+			return nil
+		},
+		GetMiniBlocksCalled: func() (map[string]*block.MiniBlock, error) {
+			return map[string]*block.MiniBlock{
+				string(mbHeaderHash2): {
+					ReceiverShardID: 1,
+					SenderShardID:   0,
+				},
+			}, nil
+		},
+	}
+
+	expectedBody := &block.Body{
+		MiniBlocks: []*block.MiniBlock{
+			{
+				ReceiverShardID: 1,
+				SenderShardID:   0,
+			},
+		},
+	}
+
+	body, err := svs.getPeerBlockBodyForMeta(metaBlock)
```

Comment:
i don't think `mbHeader` can be nil, if `peerMiniBlocks` will have not entries, blockBody.MiniBlock will be also empty

Reply:
>i don't think mbHeader can be nil,

It could be nil, since it is a slice of interfaces. One of the pointers could be nil

>if peerMiniBlocks will have no entries, blockBody.MiniBlock will be also empty

`peerMiniBlocks` is a map[string]*SomeStruct. Theoretically, It could be that there is no entry in the map for the given string.

Nothing to be done in this PR about it, just pointing out. Will leave this comm open to track left changes to be addressed